### ISSUE TITLE
Some context menus in edge have become inaccessible with NVDA

### ISSUE THREAD

**Issue by rperez030** [nvaccess/nvda#15293]
### Steps to reproduce:

This issue affects context menus such as the one in the download section, and the menu that opens after selecting More in the Add a bookmark dialog. It doesn't seem to affect Narrator or JAWS. 

1. With Microsoft edge opened, press CTRL +J to go to downloads. Make sure to have at least one element in the list and that NVDA switches to focus mode.
2. Press the applications key to open the context menu.

### Actual behavior:

The context menu opens, and the option Open file receives focus, but NVDA doesn't announce anything through Braille or speech.

### Expected behavior:

The context menu will open and the option with focus will be announced by NVDA.

### NVDA logs, crash dumps and other attachments:

This is a log fragment with MSAA and UIA both enabled in the debug categories.
 
[LogWithEdge.txt](https://github.com/nvaccess/nvda/files/12337077/LogWithEdge.txt)

#### NVDA installed/portable/running from source:

Installed.

#### NVDA version:

Version: 2023.2beta3 (2023.2.0.28878.

#### Windows version:

Edition	Windows 11 Enterprise

Version	22H2

OS build	22621.2134

#### Name and version of other software in use when reproducing the issue:

Microsoft Edge
Version 116.0.1938.43 (Official build) stable app, beta channel (arm64)

I want to clarify that I'm also able to reproduce this in the stable AMD64 version of edge.

#### Does the issue still occur after restarting your computer?

Yes

#### If NVDA add-ons are disabled, is your problem still occurring?

Yes.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

Yes.

**Comment by leonardder** [comment]
Confirmed. This bug is actually pretty annoying. It might be because we're forcefully ignoring UIA in an area where we shouldn't.

**Comment by seanbudd** [comment]
Closing as duplicate of #14916

**Comment by beqabeqa473** [comment]
I've fixed this temporarily in msedgeDiscardAnnouncements addon.

On 8/15/23, Sean Budd ***@***.***> wrote:
> Closed #15293 as not planned.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/15293#event-10092607090
> You are receiving this because you are subscribed to this thread.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by leonardder** [comment]
@beqabeqa473 Great that you were able to fix it in your add-on, but this would be a fix that might be very beneficial for NVDA core could you please consider providing a pull request to fix it?

**Comment by beqabeqa473** [comment]
I would do a pullrequest, but the way i fixed it in addon is to tell
to eventHandler to accept gainFocus event from a process and
windowclass of the nearest ancestor.

I am not sure it is an elegant solution.

On 8/15/23, Leonard de Ruijter ***@***.***> wrote:
> @beqabeqa473 Great that you were able to fix it in your add-on, but this
> would be a fix that might be very beneficial for NVDA core could you please
> consider providing a pull request to fix it?
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/15293#issuecomment-1679259306
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili
